app/controllers/users.rb

```diff
@@ -0,0 +1,39 @@
+enable :sessions
+
+post '/login' do
+  user = User.find_by(username: params[:username])
+  if user != nil && user.authenticate(params[:password])
+    session[:username] = user.username
```

Comment:
just store the user_id in the session. NEVER store the password in the session. Also all this other stuff should be looked up from the user record via the `current_user` helper